src/core/raster/qgsexiftools.cpp

```diff
@@ -18,80 +18,147 @@
+       key == QLatin1String( "Exif.GPSInfo.GPSDestLatitude" ) ||
+       key == QLatin1String( "Exif.GPSInfo.GPSDestLongitude" ) )
   {
-    const QString key = QString::fromStdString( i->key() );
-    QVariant val;
-    switch ( i->typeId() )
+    val = readCoordinate( it->value() );
+  }
+  else if ( key == QLatin1String( "Exif.GPSInfo.GPSTimeStamp" ) )
+  {
+    const QStringList parts = QString::fromStdString( it->toString() ).split( QRegularExpression( QStringLiteral( "\\s+" ) ) );
+    if ( parts.size() == 3 )
+    {
+      const int hour = readRationale( it->value(), 0 );
+      const int minute = readRationale( it->value(), 1 );
+      const int second = readRationale( it->value(), 2 );
+      val = QVariant::fromValue( QTime::fromString( QStringLiteral( "%1:%2:%3" )
+                                 .arg( QString::number( hour ).rightJustified( 2, '0' ) )
+                                 .arg( QString::number( minute ).rightJustified( 2, '0' ) )
+                                 .arg( QString::number( second ).rightJustified( 2, '0' ) ), QLatin1String( "hh:mm:ss" ) ) );
+    }
+  }
+  else if ( key == QLatin1String( "Exif.GPSInfo.GPSDateStamp" ) )
+  {
+    val = QVariant::fromValue( QDate::fromString( QString::fromStdString( it->toString() ), QLatin1String( "yyyy:MM:dd" ) ) );
+  }
+  else if ( key == QLatin1String( "Exif.Image.DateTime" ) ||
+            key == QLatin1String( "Exif.Image.DateTime" ) ||
+            key == QLatin1String( "Exif.Photo.DateTimeDigitized" ) ||
+            key == QLatin1String( "Exif.Photo.DateTimeOriginal" ) )
+  {
+    val = QVariant::fromValue( QDateTime::fromString( QString::fromStdString( it->toString() ), QLatin1String( "yyyy:MM:dd hh:mm:ss" ) ) );
+  }
+  else
+  {
+    switch ( it->typeId() )
     {
       case Exiv2::asciiString:
       case Exiv2::string:
       case Exiv2::comment:
       case Exiv2::directory:
       case Exiv2::xmpText:
-        val = QString::fromStdString( i->toString() );
+        val = QString::fromStdString( it->toString() );
         break;
 
       case Exiv2::unsignedLong:
       case Exiv2::signedLong:
-        val = QVariant::fromValue( i->toLong() );
+      case Exiv2::unsignedLongLong:
+      case Exiv2::signedLongLong:
+        val = QVariant::fromValue( it->toLong() );
         break;
 
       case Exiv2::tiffDouble:
       case Exiv2::tiffFloat:
-        val = QVariant::fromValue( i->toFloat() );
+        val = QVariant::fromValue( it->toFloat() );
         break;
 
       case Exiv2::unsignedShort:
       case Exiv2::signedShort:
-        val = QVariant::fromValue( static_cast< int >( i->toLong() ) );
-        break;
-
-      case Exiv2::unsignedRational:
-      case Exiv2::signedRational:
       case Exiv2::unsignedByte:
       case Exiv2::signedByte:
-      case Exiv2::undefined:
       case Exiv2::tiffIfd:
+      case Exiv2::tiffIfd8:
+        val = QVariant::fromValue( static_cast< int >( it->toLong() ) );
```

Comment:
Is the cast here required?

Reply:
It helps providing a more meaningful value when someone checks against the resulting QVariant's type().